server/src/main/java/io/druid/curator/inventory/CuratorInventoryManager.java

```diff
@@ -122,6 +122,12 @@ public void start() throws Exception
       }
       throw e;
     }
+
+    // Block start() till PathChildrenCache is fully loaded
+    while (!containerCacheListener.doneInitializing) {
```

Comment:
can we use a CountDownLatch.await(1 minute) here instead? that way not too many logs will be printed and also it wouldn't wait any more than necessary ?